hibernate-vector/src/main/java/org/hibernate/vector/OracleVectorJdbcType.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.vector;
+
+import java.util.Arrays;
+import java.util.BitSet;
+
+import org.hibernate.dialect.Dialect;
+import org.hibernate.sql.ast.spi.SqlAppender;
+import org.hibernate.type.SqlTypes;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.JavaType;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+
+/**
+ * Specialized type mapping for generic vector {@link SqlTypes#VECTOR} SQL data type for Oracle.
+ * <p>
+ * This class handles generic vectors represented by an asterisk (*) in the format,
+ * allowing for different element types within the vector.
+ *
+ * @author Hassan AL Meftah
+ */
+public class OracleVectorJdbcType extends AbstractOracleVectorJdbcType {
```

Comment:
This allows to inherit some methods and trim down this class.

```suggestion
public class OracleVectorJdbcType extends OracleFloatVectorJdbcType {
```